build-targets-gradle-plugin/src/main/kotlin/com/faire/gradle/release/ComputeDependentProjectsTask.kt

```diff
@@ -41,34 +38,30 @@ internal abstract class ComputeDependentProjectsTask @Inject constructor(
 
   @TaskAction
   fun execute() {
-    val projectDependencyPaths = projectDependencyPaths.get().sorted()
+    val dependentProjectPaths = TreeSet<String>()
+    DependencyGraph.traverseGraph(
+        rootComponent = rootComponent.get(),
+        rootVariant = rootVariant.get(),
+        nodeCallback = { node ->
+          val projectPath = getProjectPath(node)
+          if (projectPath != null) {
+            dependentProjectPaths.add(projectPath)
+          }
+        },
+    )
+
     dependentProjectsListFile.asFile.get().apply {
       parentFile.mkdirs()
-      writeText(projectDependencyPaths.joinToString("\n"))
+      writeText(dependentProjectPaths.joinToString("\n"))
     }
   }
 
-  /** Gets all the project dependencies for a project, including transitive ones. */
-  private fun getAllDependenciesForProjects(
-      startingProjects: Set<Project>,
-      visitedProjects: Set<Project> = setOf(),
-  ): Set<Project> {
-    if (startingProjects.isEmpty()) {
-      return setOf()
+  private fun getProjectPath(variant: ResolvedVariantResult): String? {
+    val owner = variant.owner
+    return if (owner is ProjectComponentIdentifier) {
+      owner.projectPath
+    } else {
+      null
     }
-
-    val runtimeProjectDependencies = startingProjects.asSequence()
-        .flatMap { project ->
-          val runtimeClasspath = project.configurations.named(configurationName.get()).get()
-          runtimeClasspath.allDependencies.withType<ProjectDependency>()
-              // RE: Deprecation: We don't have a better path at the moment and we've asked Gradle for support!
```

Comment:
And support we got! https://gradle-community.slack.com/archives/CA745PZHN/p1744315619297459

Reply:
I wouldn't call it "support", I'm just a user like you. :-D